compiler/compiler-settings.cpp

```diff
@@ -301,8 +301,8 @@ void CompilerSettings::init() {
 
   std::string cxx_default_flags = ss.str();
 
-  incremental_linker.value_ = dynamic_incremental_linkage.get() ? cxx.get() : "ld";
-  incremental_linker_flags.value_ = dynamic_incremental_linkage.get() ? "-shared" : "-r";
+  compiler_toolchain_option.value_ = !compiler_toolchain_dir.value_.empty() ? ("-B" + compiler_toolchain_dir.value_) : "";
```

Comment:
If toolchain_dir is empty, then "" will be passed to compiler cmd line — and it will work as expected (will just invoke some default linker, probably ld), right?

Reply:
If toolchain_dir is empty, then "" will be appended to command line that invoke compiler. So yes it safe.